src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1733,8 +1754,9 @@ void Session::activate()
     }
     catch (const IntegrationException& error) {
         logger.error("Error integrating bootstrap changesets: %1", error.what());
-        on_suspended(SessionErrorInfo{error.code(), false});
+        m_suspended = true;
         m_conn.one_less_active_unsuspended_session(); // Throws
+        on_suspended(SessionErrorInfo{error.code(), false});
```

Comment:
This was actually a bug where we'd end up making the number of active unsuspended sessions underflow.